### DISCUSSION TITLE
Trouble navigating to different screens back and forth

### DISCUSSION THREAD

**Discussion by kyusungpark** [expo/router#908] [Q&A]
Hi, I have been having some issues with redirecting to a different url back and forth.

I want to be able to navigate between two urls.
1. /students/:studentId
2. /classes/:classId

In Classes screen, I have a list of students and clicking an item should redirect to `students/:studentId`.
Same goes for Students screen, I have a list of students and clicking an item should redirect to `classes/:classesId`

Now, when i am on Classes screen and click on an item to go to `/students/:studentId` it redirects fine.
But now that I am on Student screen and I click on one of the Class items to go to `/classes:/classes` the redirect does not work. it stays on the same page.

What's weird is that when I follow the opposite steps compared to above:

I am on Student screen and click on an item to go to `/classes/:classId` it works.
Now that I am on Class screen and click on Student item, it won't redirect to `/students/:classes`.

To summarize
1. from `/students/:studentId `to `classes/:classId` works
2. now on `classes/:classId` then to `/students/:studentId` does not work

Same thing for the other scenario
1. from `/classes/:classId `to `students/:studentId` works
2. now on `students/:studentId` then to `/classes/:classId` does not work

I did noticed that after the 1st successful redirect and then try to redirect to the other page, path did not update when I logged `GlobalSearchParams`.
 
Any one have an idea what this happens? Seems like one way redirect works, but redirecting back and forth won't work.
